viewing_party/party.py

```diff
@@ -0,0 +1,259 @@
+    return  average_rating
+    
+#Wave_TWO Part 2 
+
+def get_most_watched_genre(user_data):
+    most_populare_genre =[]
+    if len (user_data["watched"]) == 0:
+        return None 
+    for movie_genre in user_data["watched"]:
+        
+            if "genre" in movie_genre:
+                most_populare_genre.append(movie_genre["genre"])
+            most_common = max(most_populare_genre, key = most_populare_genre.count)
+            print (most_common)
+
+    return most_common
+
+            
+#Completed using a dict but wasn't my first choice
+# def get_most_watched_genre(user_data):
+#     most_popular_genre = {}
+#     if len(user_data["watched"]) == 0:
+#         return None
+#     for movie_genre in user_data["watched"]:
+#         if movie_genre ["genre"] in most_popular_genre:
+#             most_popular_genre[movie_genre["genre"]] += 1
+
+#         else:
+#             most_popular_genre[movie_genre["genre"]] = 1
+
+#     max_value_genre=max(most_popular_genre, key= most_popular_genre.get)
+#     return max_value_genre
+
+#     # https://www.programiz.com/python-programming/methods/built-in/max
+        
+
+#Wave_Three Part 1 
+
+def get_unique_watched(user_data):
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    movie_list = []
+    user_movie_set = set()
+    friend_movie_set = set()
+    for movie in user_watched: 
+        user_movie_set.add(movie["title"])
+        print(user_movie_set)
+    for friend in friends_watched:
+        for friends_watched_list in friend.values():
+                # print(friends_watched_list)
+            for movie_title in friends_watched_list:
+                    # print(movie_title)
+                    friend_movie_set.add(movie_title["title"])
+
+    potential_friends_watched_set = (user_movie_set) - (friend_movie_set) 
+    final_movie_list = []
+    for potential in potential_friends_watched_set:
+        final_movie_list.append({"title":potential})
+    return final_movie_list
+
+
+
+    
+    
+
+#Wave 3 Part 2 
+
+def get_friends_unique_watched(user_data):
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
+    movie_list = []
+    user_movie_set = set()
+    friend_movie_set = set()
+    for movie in user_watched: 
+        user_movie_set.add(movie["title"])
+        print(user_movie_set)
+    for friend in friends_watched:
+        for friends_watched_list in friend.values():
+                # print(friends_watched_list)
+            for movie_title in friends_watched_list:
+                    # print(movie_title)
+                    friend_movie_set.add(movie_title["title"])
+    
+    friends_unique_movies = []
+    friends_unique_movie_set = (friend_movie_set) - (user_movie_set)
+
+    for movie in friends_unique_movie_set:
+        friends_unique_movies.append({"title":movie})
+    return friends_unique_movies
+
+
+            
+        
+
+
+
+#Wave 4 Part1
+
+
+
+
+def get_available_recs(user_data):
```

Comment:
This is a good implementation. I want to offer a suggestion that uses fewer lines of code.

`def get_available_recs(user_data):
    recommended_movies = []
    friends_unique_watched = get_friends_unique_watched(user_data)

    for movie in friends_unique_watched:
        if movie["host"] in user_data["subscriptions"]:
            recommended_movies.append(movie)   
    return recommended_movies`